code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -174,13 +181,23 @@ void handleConsentUpdate(@NonNull final Event event) {
 			return;
 		}
 
+		// check if the forceSync flag is set to true
+		boolean forceSync = DataReader.optBoolean(
+			consentData,
+			ConsentConstants.ConfigurationKey.CONSENT_FORCE_SYNC,
+			ConsentConstants.Defaults.FORCE_SYNC
+		);
+
 		// set the timestamp and merge with existing consents
 		newConsents.setTimestamp(event.getTimestamp());
-		consentManager.mergeAndPersist(newConsents);
+		// Only proceed with Edge event dispatch if preferences actually changed
+		if (consentManager.mergeAndPersist(newConsents) || shouldForceSync(event)) {
+			// share and dispatch the updated consents
```

Comment:
I have posted a question on the iOS PR as to why we are calling mergeAndPersist inside this condition block over there. Once we have clarity we can resolve this.

Reply:
The logic in Android is a little different because the Consents class has two "equals" functions, one checks equality for all values and the other ignores the timestamp when checking equality. The ConsentManager.mergeAndPersist uses the function which ignores timestamps to check if the values changed.